pkg/reconciler/v1alpha1/configuration/configuration.go

```diff
@@ -298,8 +300,22 @@ func (c *Reconciler) gcRevisions(ctx context.Context, config *v1alpha1.Configura
 		return err
 	}
 
+	// Sort by creation timestamp descending
+	sort.Slice(revs, func(i, j int) bool {
+		return revs[j].CreationTimestamp.Before(&revs[i].CreationTimestamp)
+	})
+
+	var expiry *metav1.Time
+
+	minRevisions := cfg.StaleRevisionMinimumGenerations
+	if int64(len(revs)) > minRevisions && minRevisions != 0 {
+		offset := cfg.StaleRevisionMinimumGenerations - 1
+		revisionCreation := revs[offset].CreationTimestamp
+		expiry = &revisionCreation
+	}
+
 	for _, rev := range revs {
-		if isRevisionStale(ctx, rev, config) {
+		if isRevisionStale(ctx, rev, config, expiry) {
```

Comment:
instead of expiry could we just limit the iteration to `for _, rev := range revs[offset:] {`?

Reply:
Done - thanks for the tip